esrally/client.py

```diff
@@ -129,7 +129,17 @@ def _is_set(self, client_opts, k):
 
     def create(self):
         import elasticsearch
-        return elasticsearch.Elasticsearch(hosts=self.hosts, ssl_context=self.ssl_context, **self.client_options)
+        instance = elasticsearch.Elasticsearch(hosts=self.hosts, ssl_context=self.ssl_context, **self.client_options)
+        if self._is_set(self.client_options, "use_api_key"):
```

Comment:
This would also enter the block if the user specified `use_api_key:false`on the client options. I noticed that we fall into this trap in other places in this class as well and I propose we change this line to:

```python
if self.client_options.get("use_api_key", False):
```

Reply:
That method is slightly misleading. It does not check if its set, it just removes the possibility of a key error. It returns the value, if present. Adding a test like so confirms this (and passes).

```
    def test_is_set(self):
        hosts = [{"host": "127.0.0.1", "port": 9200}]
        client_options = {}
        # make a copy so we can verify later that the factory did not modify it
        original_client_options = dict(client_options)

        f = client.EsClientFactory(hosts, client_options)

        opts = {"foo": True, "bar": False}
        assert f._is_set(opts, "foo") is True
        assert f._is_set(opts, "bar") is False
        assert f._is_set(opts, "baz") is False
```